pyActigraphy/io/cwa/cwa_class.py

```diff
@@ -1,237 +1,513 @@
+import os
 from datetime import datetime
-from struct import Struct
+from datetime import timedelta
+from tools import DataReader
+
+import pandas
+from numpy.linalg import norm
+
+class CWA(object):
+    __slots__ = ["__endianess", "__stream", "__meta",
+                 "__blocks",
+                 "__device",
+                 "__name", "__uuid", "__startTime",
+                 "__duration", "__period",
+                 "__Accel", "__Gyro", "__Mag",
+                 "__Ascale", "__Gscale", "__Mscale",
+                 "__Others"]
+    __hardware = {0x00: "AX3",
+                  0xff: "AX3",
+                  0x17: "AX3",
+                  0x64: "AX6"}
+    __axes = {1: "Axyz",
+              2: "Gxyz/Axyz",
+              3: "Gxyz/Axyz/Mxyz"
+              }
+
+    @classmethod
+    def IsValidFile(cls, filename):
+        """
+        Checks if given file is a valid cwa file.
+        Checks is performed by name (do it have .cwa)
+        and by leading bits, which must be 'MD'
+
+        Parameters
+        ----------
+        filename: str
+            path to an existing file
+
+        Returns
+        -------
+        bool
+            True if given file is valid cwa file
+            False otherwise
+
+        Raises
+        ------
+        TypeError
+            if passed parameters are of incompatible type
+        """
+
+        if not isinstance(filename, str):
+            raise TypeError("filename must be a string")
+        ext = os.path.splitext(filename)
+        if ext[1] != ".cwa":
+            return False
+
+        f = open(filename, "rb")
```

Comment:
@nbeliy : use `with open(filename, 'rb') as f:` instead. If the opening crashes (or the code crashes without reaching your `close`statement), the file will be closed anyway.

Reply:
with statement will close the file automatically at the end of block, and I want to keep the stream open